controllers/posts.js

```diff
@@ -0,0 +1,48 @@
+const Post = require("../models/post");
+
+const PostsController = {
+  Index: (req, res) => {
+    Post.find({})
+    .sort({ _id: -1 })
+    .exec(function (err, userposts) {
+      if (err) {
+        throw err;
+      }
+
+      res.render("posts/index", { posts: userposts });
+    });
+},
+  
+  New: (req, res) => {
+    res.render("posts/new", {user: req.session.user});
+  },
+  Create: (req, res) => {
+        const post = new Post(req.body);
+        post.save((err) => {
+          if (err) {
+            throw err;
+          }
+          res.status(201).redirect("/posts");
+        });
+      },
+
+  UpdateLikes: (req, res) => {
+    console.log("Update likes in controller");
+    console.log(req.params.id);
+    const action = req.body.action;
+    const counter = (action === "Like" ? 1 : -1);
```

Comment:
Good Like / Unlike feature, but currently can use refresh to keep adding likes, if had more time you could consider to record likes in the database so that likes are unique